drug_learning/two_dimensions/Input/fingerprints.py

```diff
@@ -17,6 +17,9 @@ def transform(self):
         fts = []
         self.mol_names = []
         for mol in self.structures:
+            if mol is None:
+                # skip problematic molecules
```

Comment:
@cescgina  Could we add these lines as a function of the base_class.py and then filter incorrect molecules when fitting so we do not have to repeat that much code??

Reply:
The reason this is repeated is because fitting an sdf file does nothing more than creating the SDMolSupplier object, so processing of the molecule is delayed. I think the cleanest way would be wrap this object in a generator (or list, generator would be better to reduce memory footprint) similar to the mol2 case.